### DISCUSSION TITLE
Sending and recording S21 messages via Daikin's wired remote?

### DISCUSSION THREAD

**Discussion by ThiefMaster** [revk/xESP32-Faikout#117] [General]
Daikin sells a wired remote (BRC073) that connects to the S21 port.

Do you think it would be feasible to support some kind of pass-through mode so you can connect the Faikin to the Daikin and the wired remote to the Faikin, in order to:

- record S21 messages for various actions done on the wired remote (I think this might be the easiest way to get all available functionality into Faikin
- use both Faikin and the wired remote (since there's probably only one S21 port so right now it'd have to be either Faikin or the wired remote)

I think connecting both is likely a bit tricky, unless the faikin can simply "monitor" the S21 bus. But if you think there is a simple way (making a custom Y cable to connect faikin and the remote and/or flashing custom firmware on the Faikin would be OK for me) I would seriously consider buying a wired remote and providing S21 recordings from that :)

**Comment by revk** [comment]
There is C code to run in Linux to act as the aircon and record the remote S21 logic.

**Reply by trevorbattle** [reply]
Link?

**Reply by revk** [reply]
Tools/faikin.c

**Reply by ThiefMaster** [reply]
Isn't this limited to S21 messages where you know what they do (and when they should be sent) though? That's why I mentioned the wired remote since AFAIU it is the most feature-complete way to control the AC (and get all kinds of interesting status information), so it could provide a great way to sniff any commands that aren't supported yet...

**Reply by revk** [reply]
Yes, but easy to expand for more messages

**Reply by ThiefMaster** [reply]
exactly - so I guess using a remote to record the missing stuff would nonetheless be useful. So not really knowing much of the S21 bus protocol my main question still stands:

Can I have two things connected at the same time to S21 (Wired remote to send/receive commands from the AC + Faikin to sniff the communication that's going on)? Or do you see another way how this could be done (e.g. connecting the remote to the faikin and logging what it sends, then replaying it to the AC and logging the reply)

**Reply by revk** [reply]
I did create a monitor once, and may be able to make again. I'll have to did how I did it.
